src/App.tsx

```diff
@@ -1,222 +1,203 @@
+  useEffect,
+  // useMemo,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import {
+  getTodos,
+  createTodo,
+  removeTodo,
+  changeTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { FilterCondition, ErorrMessage } from './types/enums';
+import { Header } from './components/Main/Header';
+import { TodoList } from './components/Main/TodoList';
+import { Footer } from './components/Main/Footer';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+  const [filterCondition, setFilterCondition] = useState(FilterCondition.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempNewTask, setTempNewTask] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+
+  if (isError) {
+    setTimeout(() => setIsError(false), 3000);
+  }
+
+  const getSelected = useCallback(
+    (allTodos: Todo[]): Todo[] => {
+      switch (filterCondition) {
+        case FilterCondition.COMPLETED:
+          return allTodos.filter(item => item.completed === true);
+
+        case FilterCondition.ACTIVE:
+          return allTodos.filter(item => item.completed === false);
+
+        case FilterCondition.ALL:
+        default:
+          return allTodos;
+      }
+    }, [filterCondition],
+  );
+
+  const addTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    const fullTodoData = { ...todoData, userId: user?.id };
+
+    try {
+      const temporaryTodo = {
+        ...todoData,
+        id: 0,
+      };
+
+      setTempNewTask(temporaryTodo);
+
+      setIsAdding(true);
+      const newTodo = await createTodo(fullTodoData);
+
+      setTodosList(currentTodos => [...currentTodos, newTodo]);
+    } catch {
+      setErrorText(ErorrMessage.ON_ADD);
+    } finally {
+      setTempNewTask(null);
+      setIsAdding(false);
+    }
+  }, [user?.id]);
+
+  const deleteTodo = useCallback(async (idToDelete: number) => {
+    try {
+      setDeletingTodoIds(curr => [...curr, idToDelete]);
+
+      await removeTodo(idToDelete);
+
+      setTodosList((currentTodos: Todo[]) => currentTodos
+        .filter(task => task.id !== idToDelete));
+    } catch {
+      setIsError(true);
+      setErrorText(ErorrMessage.ON_DELETE);
+    } finally {
+      setDeletingTodoIds(currId => currId.filter(id => id !== idToDelete));
+    }
+  }, []);
+
+  const getCompletedTodos = (allTodos: Todo[]) => {
+    const completedTodos = allTodos.filter(todo => todo.completed === true);
+
+    return completedTodos.map(todo => todo.id);
+  };
```

Comment:
The function name looks like this function return todos, but the function return ids, let's fix name or logic

Reply:
+